src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyNameHelpers.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Collections.Generic;
+
+namespace System.Reflection
+{
+    internal static partial class AssemblyNameHelpers
+    {
+        //
+        // These helpers convert between the combined flags+contentType+processorArchitecture value and the separated parts.
```

Comment:
Are these helpers actually needed in the shared CoreLib after the last set of changes?

Reply:
`ExtractProcessorArchitecture` and `CombineAssemblyNameFlags` are only used by nativeaot now, 
Other two have users in `MetadataLoadContext`

Those existed before, I just noticed that we duplicate these helpers, so I merged them.

Reply:
These helpers are in CoreLib now, but the Corelib does not actually use them. 

I think it is fine to have them duplicated between NativeAOT reflection and MetadataLoadContext.